### PR TITLE
[enh] Certificate management interface

### PR DESCRIPTION
**Pull request by alexAubin**
Hi guys,

This pull requests adds a new interface to manage the certificate of domains, using the certificate management feature ([ongoing PR on yunohost CLI](https://github.com/YunoHost/yunohost/pull/180)).

Below are a few screenshots of the main features / buttons.

I hope my code isn't too dirty - I'm not used to dev this kind of interface. Feel free to request tweaks or other or features you'd like to see added for this interface.

### New button on 'domain info' page
screenshot

### Cert management page for a domain with self-signed cert
screenshot

### Cert management page for a domain with let's encrypt cert
screenshot

### CODE REVIEW

File: src/js/yunohost/controllers/domains.js [hunk 1]
```diff
@@ -81,12 +81,26 @@
...
+                });
+
+
+                domain = {
+                    name: c.params['domain'],
+                    main: (c.params['domain'] == dataMain.current_main_domain) ? true : false,
+                    url: "https://"+c.params['domain'],
+                    enable_cert_management: enable_cert_management_
+                };
+                console.log(domain);
```
Comment (opi) [inline src/js/yunohost/controllers/domains.js]: remove console.log

File: src/views/domain/domain_info.ms [hunk 2]
```diff
@@ -57,6 +57,19 @@
...
         <div class="container">
+            <p>Gérer le certificat</p>
+            {{#unless enable_cert_management}}
+            <p><span class="fa-fw fa-exclamation-circle"></span> The 'letsencrypt'
+            app is currently installed and conflicts with this feature. Please
+            uninstall it first to use the new certificate management
+            interface.</p>
+            {{/unless}}
+            <a href="#/domains/{{name}}/cert-management" class="btn btn-default slide {{#unless enable_cert_management}}disabled{{/unless}}">
+                Certificat SSL <span class="fa-fw fa-lock"></span>
```
Comment (opi) [inline src/views/domain/domain_info.ms]: untranslated french string

**Comment by opi** [comment]
Made a first review, and everything looks good. Thanks for this awesome piece of work, I'm eager to test it.

**Comment by M5oul** [comment]
I encounter an alignment issue on the left:

ekrankopio je 2016-11-25 23 00 54-fullpage

**Comment by M5oul** [comment]
```bash
Error: 404 Not Found

Sorry, the requested URL 'http://127.0.0.1:6787/domains/cert-status/le.moul.re?full&locale=en' caused an error:

Not found: '/domains/cert-status/le.moul.re'
```

**Comment by opi** [comment]
> I encounter an alignment issue on the left:

this is due to an error returned by the API (Mostly your later 404). It sometimes breaks the admin layout. This is not really related to this PR

**Comment by M5oul** [comment]
> this is due to an error returned by the API (Mostly your later 404). It sometimes breaks the admin layout. This is not really related to this PR

Nice

----

> A` service yunohost-api reload` was needed.

Do this will be needed on a fresh install?

**Comment by opi** [comment]
https://github.com/yunohost/yunohost/blob/unstable/debian/rules#L11 If I understand this line correctly, the *yunohost-api* service is reload during debian package upgrade; I guess (and hope) you had this error only because you are testing without packaging (by using `ynh-dev use-git` command)

**Comment by Psycojoker** [comment]
Yes, the api is reloaded on upgrade (this is why the upgrade is delegated in a cron file) and if you change python files you need to reload.

File: src/locales/en.json [hunk 3]
```diff
@@ -294,5 +294,30 @@
...
+    "confirm_cert_install_LE": "Are you sure you want to install a Let's Encrypt certificate for this domain.",
+    "confirm_cert_regen_selfsigned": "Are you sure you want to regenerate a self-signed certificate for this domain ?",
+    "confirm_cert_manual_renew_LE": "Are you sure you want to manually renew the Let's Encrypt certificate for this domain now ?",
+    "confirm_cert_revert_to_selfsigned": "Are you sure you want to revert this domain to a self-signed certificate ?",
+    "certificate" : "Certificate",
+    "certificate_status" : "Certificate status",
+    "certificate_authority" : "Certification authority",
+    "validity" : "Validity",
+    "domain_is_eligible_for_ACME" : "This domain seems correctly configured to install a Let's Encrypt certificate !",
+    "domain_not_eligible_for_ACME" : "This domain doesn't seem ready for a Let's Encrypt certificate. Please check your DNS configuration and HTTP server reachability.",
```
Comment (Psycojoker) [inline src/locales/en.json]: A decicated page would be better, but I don't know if someone would be willing to do it.

**Comment by alexAubin** [comment]
I think this is a medium decision. We need one more approval for this ?

**Comment by zamentur** [comment]
There is 4 positive opinions. So conditions to merge are respected: > 50% and the PR don't change since 2 weeks.

**Pull request merged**